source/README.adoc

```diff
@@ -0,0 +1,171 @@
+| `id` | A unique ID | This uses the `MessageId` if available, and a UUID otherwise
+| `subject` | The ID of the message | Empty string if no message ID is present
+| `datacontenttype` | `application/json` | Currently static
+| `key` | The ID of the message | Empty string if no message ID is present
+|===
+
+The payload of the event is set to the data content of the message.
+
+=== Samples
+
+For a message published with the payload "Hello rabbitmq!", for example with
+https://www.rabbitmq.com/management-cli.html[`rabbitmqadmin`]:
+
+[source,shell]
+----
+rabbitmqadmin publish exchange=amq.default payload="Hello rabbitmq!"
+----
+
+The source sends the following event content:
+
+.CloudEvents JSON format
+[source,json]
+----
+{
+  "specversion": "1.0",
+  "type": "dev.knative.rabbitmq.event",
+  "source": "/apis/v1/namespaces/default/rabbitmqsources/rabbitmq-source",
+  "id": "f00c1f52-33a1-4d3d-993f-750f20c804da",
+  "time": "2020-12-18T01:15:20.450860898Z",
+  "subject": "",
+  "datacontenttype": "application/json",
+  "key": "",
+  "data": "Hello rabbitmq!"
+}
+----
+
+== Creating and managing sources
+
+Sources are Kubernetes objects. In addition to the standard Kubernetes
+`apiVersion`, `kind`, and `metadata`, they have the following `spec` fields:
+
+:optional: *
+
+.Source parameters
+|===
+| Field | Value
+
+| `spec.brokers` | Host+Port of the Broker, with a trailing "/"
+| `user.secretKeyRef` | Username for Broker authentication; field `key` in a Kubernetes Secret named `name`
+| `password.secretKeyRef` | Password for Broker authentication; field `key` in a Kubernetes Secret named `name`
+| `topic` | The topic for the exchange
+| `exchange_config` | Settings for the exchange
+| `exchange_config.type` | https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchanges[Exchange type]. Can be `fanout`, `direct`, `topic`, `match` or `headers`
+| `exchange_config.durable` {optional} | Boolean
+| `exchange_config.auto_deleted` {optional} | Boolean
+| `exchange_config.internal` {optional} | Boolean
+| `exchange_config.nowait` {optional} | Boolean
+| `queue_config` {optional} | Settings for the queue
+| `queue_config.name` {optional} | Name of the queue (may be empty)
+| `queue_config.routing_key` {optional} | Routing key for the queue
+| `queue_config.durable` {optional} | Boolean
+| `queue_config.delete_when_unused` {optional} | Boolean
+| `queue_config.exclusive` {optional} | Boolean
+| `queue_config.nowait` {optional} | Boolean
+| `sink` | A reference to an https://knative.dev/docs/eventing/#event-consumers[Addressable] Kubernetes object
+|===
+
+{optional} These attributes are optional.
+
+You will need a Kubernetes Secret to hold the RabbitMQ username and
+password. The following command is one way to create a secret with the username
+`rabbit-user` and the password taken from the `/tmp/password` file.
+----
+kubectl create secret generic rabbitmq-secret \
+  --from-literal=user=rabbit-user \
+  --from-file=password=/tmp/password
+----
```

Comment:
This is buried for something that looks important, I don't like the structure of this "Creating and managing sources" section, because a user will come here looking for a clear procedure and instead it's a lot of confusing reference type info

Reply:
Each source needs some credentials (username/password) for connecting to the RabbitMQ instance. Something along these lines will be pretty typical.

I can change the second sentence to "with `kubectl`:" or "For example:"; there are a lot of different ways the secret *could* be populated, and we've got this in here simply to show an easy way to kick the tires.